### ISSUE TITLE
Setting the same key name "`x`" in an extended `TypedDict` class as the same parameter name "`x`" gets the wrong error message saying `argument` but not `parameter`

### ISSUE THREAD

**Issue by hyperkai** [python/mypy#20933]
*Notes:
- mypy test.py
- mypy 1.19.1
- Python 3.14.0

Setting the same key name "`x`" in an extended [TypedDict](https://docs.python.org/3/library/typing.html#typing.TypedDict) class as the same parameter name "`x`" gets the wrong error message saying `argument` but not `parameter` as shown below:

```python
from typing import TypedDict, Unpack

class MyDict(TypedDict):
    x: str
    y: bool

def func(x: int, **kwargs: Unpack[MyDict]): ...
# Error
```

> error: Overlap between argument names and ** TypedDict items: "x"

So the error message should say `parameter` as shown below:

> error: Overlap between parameter names and ** TypedDict items: "x"
